features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/FlagsClient.kt

```diff
@@ -77,109 +108,351 @@ interface FlagsClient {
+        fun build(): FlagsClient {
+            // Validate that the Flags feature is enabled
+            val flagsFeature = (sdkCore as? FeatureSdkCore)
+                ?.getFeature(FLAGS_FEATURE_NAME)
+                ?.unwrap<FlagsFeature>()
+
+            if (flagsFeature == null) {
+                return NoOpFlagsClient(
+                    name = name,
+                    reason = "Flags feature not enabled",
+                    logger = (sdkCore as? FeatureSdkCore)?.internalLogger
+                )
+            }
+
+            val key = Companion.ClientKey(sdkCore, name)
+
+            synchronized(Companion.registeredClients) {
+                // Check for existing client
+                val existingClient = Companion.registeredClients[key]
+                if (existingClient != null) {
+                    logWarning(
+                        sdkCore,
+                        "Attempted to create a FlagsClient named '$name', but one already exists. " +
+                            "Existing client will be used, and new configuration will be ignored."
+                    )
+                    return existingClient
+                }
+
+                // Merge configuration (selective override using data class copy)
+                val featureConfig = flagsFeature.flagsConfiguration
+                val mergedConfig = FlagsConfiguration(
+                    customExposureEndpoint = explicitCustomEndpoint ?: featureConfig.customExposureEndpoint,
+                    customFlagEndpoint = explicitFlaggingProxy ?: featureConfig.customFlagEndpoint,
+                    enableExposureLogging =
+                    explicitEnableExposureLogging ?: flagsFeature.flagsConfiguration.enableExposureLogging
+                )
+
+                // Create and register client
+                val newClient = Companion.createInternal(mergedConfig, sdkCore as FeatureSdkCore, flagsFeature)
+                if (newClient !is NoOpFlagsClient) {
+                    Companion.registeredClients[key] = newClient
+                }
+                return newClient
+            }
+        }
+
+        private fun logWarning(sdkCore: SdkCore, message: String) {
+            (sdkCore as? FeatureSdkCore)?.internalLogger?.log(
+                InternalLogger.Level.WARN,
+                InternalLogger.Target.USER,
+                { message }
+            )
+        }
+    }
+
     /**
      * Companion object providing static access to [FlagsClient] instances.
      *
      * This companion manages the registration and retrieval of [FlagsClient] instances
      * per SDK core, ensuring thread-safe access and proper lifecycle management.
      */
     companion object {
-        private val registeredClients: MutableMap<SdkCore, FlagsClient> = mutableMapOf()
+        /**
+         * Composite key for storing multiple named clients per SDK core.
+         */
+        private data class ClientKey(val sdkCore: SdkCore, val name: String)
```

Comment:
minor: it is better to use `sdkCore.name` as a key here, to avoid storing the reference to `SdkCore`.